### ISSUE TITLE
Option to block purchase of products by category

### ISSUE THREAD

**Issue by crashzk** [Azuriom/Plugin-Shop#199]
An option for brands would be interesting if we want to block users from purchasing from this category if they already have a product from it.

Example, I create the VIP category and add the products there.
![image](https://github.com/Azuriom/Plugin-Shop/assets/32937653/a679c70f-99bb-4752-b707-ef2cfadccd6a)

In this VIP category, the user will only be able to purchase one product at a time and will only be able to purchase another when the product he purchased expires.

Same thing for selecting the product, if he has added Product 1 from the VIP category to the cart, he will only be able to add another product from the VIP category if he removes the one he already has in the cart, that is, adding several products to the cart from the same category will not it would be possible.

**Comment by MrMicky-FR** [comment]
Please submit this feedback in #31, as expiring products aren’t added in the shop plugin outside of beta

**Comment by crashzk** [comment]
Made @MrMicky-FR, but it would be perfect for products that are not in the subscription category as well, for one-time purchases.

**Comment by MrMicky-FR** [comment]
> Made @MrMicky-FR, but it would be perfect for products that are not in the subscription category as well, for one-time purchases.

Yes but currently one time purchases cannot be expiring, will be introduced with the subscriptions update
